src/Pagination.js

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import classNames from 'classnames';
+import BootstrapMixin from './BootstrapMixin';
+import PaginationButton from './PaginationButton';
+
+const Pagination = React.createClass({
+  mixins: [BootstrapMixin],
+
+  propTypes: {
+    activePage: React.PropTypes.number,
+    items: React.PropTypes.number,
+    maxButtons: React.PropTypes.number,
+    ellipsis: React.PropTypes.bool,
+    first: React.PropTypes.bool,
+    last: React.PropTypes.bool,
+    prev: React.PropTypes.bool,
+    next: React.PropTypes.bool,
+    onSelect: React.PropTypes.func
+  },
+
+  getDefaultProps() {
+    return {
+      activePage: 1,
+      items: 1,
+      maxButtons: null,
+      first: false,
+      last: false,
+      prev: false,
+      next: false,
+      ellipsis: true,
+      bsClass: 'pagination'
+    };
+  },
+
+  handleSelect(event, selectedEvent) {
+    if(this.props.onSelect){
+      this.props.onSelect(event, selectedEvent);
+    }
+  },
+
+  renderPageButtons() {
+    let pageButtons = [];
+    let startPage, endPage, hasHiddenPagesBefore, hasHiddenPagesAfter;
+    let visiblePagesBefore = parseInt(this.props.maxButtons / 2);
```

Comment:
Since you keep needing `this.props` all over in this function you can DRY this up a little bit with:

```
let {
  maxButtons,
  activePage,
  items
} = props;
```